src/Illuminate/Support/Url.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Stringable;
+
+class Url implements Arrayable, Stringable
+{
+    /**
+     * Constructor.
+     */
+    public function __construct(
+        public ?string $scheme = null,
+        public ?string $host = null,
+        public ?string $port = null,
+        public ?string $user = null,
+        public ?string $pass = null,
+        public ?string $path = null,
+        public ?string $query = null,
+        public ?string $fragment = null,
+    ) {
+    }
+
+    /**
+     * Parse a URL string into a URL object.
+     *
+     * @param  string  $url
+     * @return static
+     */
+    public static function parse(string $url): static
+    {
+        $components = parse_url($url) ?? [];
```

Comment:
The `?? []` probably doesn't catch all that one might expect it to catch as `parse_url('http:///example.com')` returns `false`. I don't know if we want to throw an error here or just always fall back to an empty array (`?:` might do it), but having `$components` sometimes equal `false` would set up a trap for whoever edits this.

Reply:
Thanks @tontonsb! Will patch 🙏